graphql/execute.lua

```diff
@@ -301,30 +301,34 @@ local function getFieldEntry(objectType, object, fields, context)
   }
 
   local resolvedObject, err = (fieldType.resolve or defaultResolver)(object, arguments, info)
-  if err ~= nil then
+  if resolvedObject == nil and err ~= nil then
     error(err)
   end
 
   local subSelections = mergeSelectionSets(fields)
   return completeValue(fieldType.kind, resolvedObject, subSelections, context,
     {fieldName = fieldName}
-  )
+  ), err
 end
 
 evaluateSelections = function(objectType, object, selections, context)
   local result = {}
+  local errors = {}
+  local err
   local fields = collectFields(objectType, selections, {}, {}, context)
   for _, field in ipairs(fields) do
     assert(result[field.name] == nil,
       'two selections into the one field: ' .. field.name)
-    result[field.name] = getFieldEntry(objectType, object, {field.selection},
+    result[field.name], err = getFieldEntry(objectType, object, {field.selection},
                                        context)
-
+    if err ~= nil then
+        table.insert(errors, err)
+    end
     if result[field.name] == nil then
         result[field.name] = box.NULL
     end
   end
-  return result
+  return result, errors
```

Comment:
I think it could break existing systems. Since some tests and projects use

```
if err ~= nil then
...
end
``` 

or

```
assert 'errors' not in result
```

In your case you always return at least empty array of errors. And it's not OK.
Moreover in the spec you refer it's mentioned:
```
The errors entry in the response is a **non‐empty** list of errors, where each error is a map.
```

I suggest to use following pattern

```
local errors

if err ~= nil then
    errors = errors or {}
    table.insert(errors, err)
end
```

Reply:
Fixed backward compatibility and added a sipmle test